src/Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -43,7 +43,7 @@ public virtual async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationT
 
             while (true)
```

Comment:
Should cancellation checked here too?

I.e.
```c#
while (!cancellationToken.IsCancellationRequested)
{
    // ....
}
```

Same below on line 84.

Reply:
Eh, then I'd have to throw in if `cancellationToken.ThrowIfCancellationRequested()`. I'd also need to handle the case where the loop never ran and the token got triggered.

Reply:
Makes sense -- I missed the throwing-behavior. 

In
https://github.com/aspnet/KestrelHttpServer/blob/105c395bac37186b789496a5656848bf318eb8db/src/Kestrel.Core/Internal/Http/MessageBody.cs#L104
the token could be passed in too. (Can't comment direct on that line here).

Reply:
good catch